From 8158e45455f507011b54b57d9655364f273e4a1c Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Tue, 22 Nov 2016 18:15:21 +0200
Subject: [PATCH 0603/1345] dts: a8040: enable phy initialization GOP2 at CP1
 device tree

commit  61e704a9947c5b27fb912aa22020398ca110cb8f from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- add mdio node to CP1 device tree
- connect CP1 GOP2 port to the CP1 mdio
- to handle CP1 port phy, mdio controller driver should be
  configured in device tree

Change-Id: Icc3428b2f30ad165beb183fd8cb7002fd9ab6cc2
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33966
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
index e84a9b1..d9bfeb6 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
@@ -130,6 +130,13 @@
 			sata@540000 {
 				status = "okay";
 			};
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy2_1: ethernet-phy@2 {
+					reg = <0>;
+				};
+			};
 			gop {
 				emac0_1: mac0 {
 					phy-mode = "sfi"; /* lane-2 */
@@ -138,7 +145,7 @@
 				};
 				emac2_1: mac2 {
 					phy-mode = "rgmii"; /* rgmii-0 */
-					phy = <&phy2>;
+					phy = <&phy2_1>;
 				};
 				emac3_1: mac3 {
 					phy-mode = "sgmii"; /* lane-4 */
-- 
1.7.9.5

